crates/iota-indexer/src/indexer_reader.rs

```diff
@@ -632,18 +632,44 @@ impl IndexerReader {
             .iter()
             .map(|digest| digest.inner().to_vec())
             .collect::<Vec<_>>();
-        run_query!(&self.pool, |conn| {
+        let checkpointed_txs = run_query!(&self.pool, |conn| {
             transactions::table
                 .inner_join(
                     tx_digests::table
                         .on(transactions::tx_sequence_number.eq(tx_digests::tx_sequence_number)),
                 )
                 // we filter the tx_digests table because it is indexed by digest,
                 // transactions table is not
-                .filter(tx_digests::tx_digest.eq_any(digests))
+                .filter(tx_digests::tx_digest.eq_any(&digests))
                 .select(StoredTransaction::as_select())
                 .load::<StoredTransaction>(conn)
-        })
+        })?;
+        let missing_digests: Vec<Vec<u8>> = {
+            let loaded_digests: HashSet<_> = checkpointed_txs
+                .iter()
+                .map(|tx| tx.transaction_digest.clone())
+                .collect();
+            digests
+                .into_iter()
+                .filter(|d| !loaded_digests.contains(d))
+                .collect()
+        };
```

Comment:
Two remarks here:

1. We can return early if `checkpointed_txs.len() = digests.len()`
2. This can be simplified and optimized if `digests` is collected as a `HashSet` instead.

Reply:
Done